assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -11508,8 +11508,8 @@ public boolean readGradeForm(RunData data, SessionState state, String gradeOptio
 
             String privateNotes = processFormattedTextFromBrowser(state, params.getCleanString(GRADE_SUBMISSION_PRIVATE_NOTES), true);
             // private notes value changed?
-            hasChange = submission != null && valueDiffFromStateAttribute(state, privateNotes, submission.getPrivateNotes());
-            if (feedbackComment != null) {
+            hasChange = !hasChange && submission != null ? valueDiffFromStateAttribute(state, privateNotes, submission.getPrivateNotes()) : hasChange;
```

Comment:
Do we apply same change here: https://github.com/sakaiproject/sakai/pull/11453/files#diff-6c59c84e25a593257fed1b6890a60f654495573e4366cf3a181f8fd1fc0da5c9R11504?

Reply:
no need as that is the first condition of hasChanged